src/Drupal/DrupalExtension/Context/EntityContext.php

```diff
@@ -0,0 +1,89 @@
+<?php
+
+namespace Drupal\DrupalExtension\Context;
+
+use Behat\Behat\Context\TranslatableContext;
+use Behat\Mink\Element\Element;
+
+use Behat\Gherkin\Node\TableNode;
+
+/**
+ * Provides pre-built step definitions for testing any Drupal entity.
+ */
+class EntityContext extends RawDrupalContext implements TranslatableContext
+{
+
+  /**
+   * Returns list of definition translation resources paths.
+   *
+   * @return array
+   */
+  public static function getTranslationResources()
+  {
+    return glob(__DIR__ . '/../../../../i18n/*.xliff');
+  }
+
+  /**
+   * @Given I am viewing a/an :entity_type entity/item with the :label_name :label
+   * @Given I am viewing a/an :bundle :entity_type entity/item with the :label_name :label
+   */
+  public function createEntity($entity_type, $bundle = NULL, $label_name, $label)
```

Comment:
I don't like this from a BDD perspective, this is using the machine name of the label property in the step definition. The machine names are not usable for several reasons: first of all they often don't match the domain specific language, they might contain abbreviations and special symbols like underscores, and they are usually in English so including machine names breaks translatability.

Also, not all entities have a label, this is not a required base field. I just had a look and actually it seems like entities with labels are in the minority, most do not have one. We cannot determine it programmatically either because there is no entity key for the label.

I would drop this step.